### PR TITLE
De widget

### PR DESCRIPTION
**Pull request by antazoey**

### CODE REVIEW

File: Packs/Code42/Scripts/Code42GetDepartingEmployees/Code42GetDepartingEmployees.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import demistomock as demisto
```
Comment (antazoey) [inline Packs/Code42/Scripts/Code42GetDepartingEmployees/Code42GetDepartingEmployees.py]: The automation script that feeds the widget

File: Packs/Code42/Scripts/Code42GetDepartingEmployees/Code42GetDepartingEmployees.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+res = {"total": 0, "data": []}
+
+try:
+    employees = demisto.executeCommand("code42-departingemployee-get-all", {})[0]["Contents"]
+    res["total"] = len(employees)
+
+    for employee in employees:
+        username = employee["userName"]
+        alerts = demisto.executeCommand("code42-alert-search", {"username": username})[0]["Contents"]
```
Comment (antazoey) [inline Packs/Code42/Scripts/Code42GetDepartingEmployees/Code42GetDepartingEmployees.py]: It's really slow, it'd be ideal to parallelize this

File: Packs/Code42/Widgets/widget-Departing_Employees_Alerts.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (antazoey) [inline Packs/Code42/Widgets/widget-Departing_Employees_Alerts.json]: Widget that displays each employee with alerts

File: Packs/Code42/Scripts/Code42GetDepartingEmployeeAlerts/Code42GetDepartingEmployeeAlerts.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import demistomock as demisto
```
Comment (unparalleled-js) [inline Packs/Code42/Scripts/Code42GetDepartingEmployeeAlerts/Code42GetDepartingEmployeeAlerts.py:1]: Not yet sure how or if this needs unit tests

File: Packs/Code42/Integrations/Code42/Code42.py [hunk 5]
```diff
@@ -509,6 +513,27 @@ def alert_resolve_command(client, args):
         return_error(create_command_error_message(demisto.command(), e))
 
 
+def alert_search_command(client, args):
+    username = args["username"]
+    try:
+        alerts = client.search_alerts(username)
+        alert_context = []
+        for alert in alerts:
```
Comment (kiran-chaudhary) [inline Packs/Code42/Integrations/Code42/Code42.py:521]: room for list comprehension....

`alert_context = [map_to_code42_alert_context(alert) for alert in alerts]`

File: Packs/Code42/Scripts/Code42GetDepartingEmployeeAlerts/Code42GetDepartingEmployeeAlerts.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+        # Sort such that highest alert counts are first and get top.
+        res["data"] = sorted(res_data, key=lambda x: x["Alerts Count"], reverse=True)[:top]
+        demisto.results(res)
+    except Exception as e:
+        res["total"] = -1
+        res["data"] = str(e)
+
+    # Submit final results to Cortex XSOAR
+    demisto.results(res)
```
Comment (kiran-chaudhary) [inline Packs/Code42/Scripts/Code42GetDepartingEmployeeAlerts/Code42GetDepartingEmployeeAlerts.py:33]: Do we need line 27,  seems redundant, `results` method is called twice.

File: Packs/Code42/Scripts/Code42GetDepartingEmployeeAlerts/Code42GetDepartingEmployeeAlerts.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+
+def main():
+    res = {"total": 0}
+    res_data = []
+    top = demisto.args().get("top") or 10
+
+    try:
+        employees = demisto.executeCommand("code42-departingemployee-get-all", {})[0]["Contents"]
+        res["total"] = len(employees)
```
Comment (kiran-chaudhary) [inline Packs/Code42/Scripts/Code42GetDepartingEmployeeAlerts/Code42GetDepartingEmployeeAlerts.py:12]: If we are ignoring certain employees later, should we not set `total` after the for loop?

Comment (kiran-chaudhary) [reply Packs/Code42/Scripts/Code42GetDepartingEmployeeAlerts/Code42GetDepartingEmployeeAlerts.py:12]: Depending on what the count represents, if it represents all employees irrespective of whether they have alerts or not then this is ok.

**Review by kiran-chaudhary — COMMENTED** [review]
Apart from few suggestions, LGTM.

**Pull request closed without merging**